src/fluree/db/validation.cljc

```diff
@@ -263,79 +269,70 @@
-    ::graph                [:orn {:error/message "Value of \"graph\" should be ledger alias or variable"}
+                            [:enum {:error/message "unrecognized where operation, must be one of: graph, filter, optional, union, bind"}
+                             :graph :filter :optional :union :bind]]
+    ::graph                [:orn {:error/message "value of graph. Must be a ledger name or variable"}
                             [:ledger ::ledger]
                             [:variable ::var]]
-    ::graph-map            [:map {:closed true
-                                  :error/message "Named where map must be a map specifying the graph to query and valid where clause"}
-                            [:graph ::graph]
-                            [:where [:ref ::where]]]
-    ::where-map            [:orn {:error/message "Invalid where map, must be either named graph or valid where operation map"}
-                            [:named ::graph-map]
-                            [:default
-                             [:and
-                              [:map-of {:max 1 :error/message "Unnamed where map can only have 1 key/value pair"}
-                               ::where-op :any]
-                              [:multi {:dispatch where-op}
-                               [:filter [:map [:filter [:ref ::filter]]]]
-                               [:optional [:map [:optional [:ref ::optional]]]]
-                               [:union [:map [:union [:ref ::union]]]]
-                               [:bind [:map [:bind [:ref ::bind]]]]]]]]
-    ::where-tuple          [:orn {:error/message "Invalid tuple"}
-                            [:triple ::triple]
-                            [:remote [:sequential {:max 4} :any]]]
-    ::where                [:sequential {:error/message "Where must be a vector of clauses"}
-                            [:orn {:error/message "where clauses must be valid tuples or maps"}
-                             [:where-map ::where-map]
-                             [:tuple ::where-tuple]]]
+    ::node-map-key         [:orn {:error/message "node map keys must be an iri or variable"}
+                            [:iri ::iri]
+                            [:var ::var]]
+    ::node-map-value       [:orn {:error/message "node map values must be an iri, string, number, boolean, map, or variable"}
+                            [:var ::var]
+                            [:string :string]
+                            [:boolean :boolean]
+                            [:int :int]
+                            [:double :double]
+                            [:nil :nil]
+                            [:iri ::iri]
+                            [:map [:ref ::node-map]]]
+    ::node-map             [:map-of {:error/message "Invalid node map"}
+                            [:ref ::node-map-key] [:ref ::node-map-value]]
+    ::where-pattern        [:multi {:dispatch where-pattern-type}
```

Comment:
FYI right now if you have a where pattern that doesn't match any of these (eg `[:foo "(> ?age 45)"]`), you'll get malli's standard "invalid dispatch value" error. If you add an `:error/message` to the map on this `:multi`, then we could override that with something slightly more helpful

There are likely other places where messages would help too, but this one jumped out at me